src/main/java/dev/seniorjava/githubstatisticsbot/listeners/CommitChosenEventListener.java

```diff
@@ -0,0 +1,31 @@
+package dev.seniorjava.githubstatisticsbot.listeners;
+
+import static java.lang.String.format;
+
+import com.pengrad.telegrambot.TelegramBot;
+import com.pengrad.telegrambot.model.request.ForceReply;
+import com.pengrad.telegrambot.request.SendMessage;
+import dev.seniorjava.githubstatisticsbot.events.CommitChosenEvent;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class CommitChosenEventListener {
```

Comment:
Я бы рекомендовал поместить код обработчиков в один класс `EventListener` иначе он размазан на пять классов. По семантике эти события связаны, они представляют события взаимодействия с пользователем? поэтому это не должно нарушать SRP

Reply:
TBH, I don't agree. Placing all handlers inside one class potentially will violate a few principles of **SOLID**.

**1. Interface Segregation Principle**. It is a good practice to have a single interface with as few as possible methods inside to make it feature-consistent. It may look like all listeners are feature-consistent. But this is not correct. All of them may be removed, replaced, used separately or we can add new listeners to cover new functionalities. My potential mistake is that my listeners do not implement a common listener. And the reason is: because it is not necessary. Spring allows me to use simple small classes without a common parent. But let's continue 🙂

**2. Liskov Substitution Principle**. Making listeners implementing a common interface will allow us to substitute the particular implementation with another one whenever we might need this and our application should continue working but with different behavior. As I've mentioned, I didn't do this. I wanted to implement the task as simply as possible in a short time.

**3. Open-Closed Principle**. Imagine we keep all listeners in the same class. Whenever we need to add an additional listener method we need to open the class, edit it and close. This already violates this principle, let's go further. Having a few methods in the same class may lead to the temptation to extract a common (on the first look) piece of code to a private method. And it may be ok for some time. But later we need to change some behavior for one method and it will affect the second method. This situation will cause the next violation:

**4. Single Responsibility Principle** - there should never be more than one reason for a class to change. Changing a piece of code that affects two public methods means the class has more than one responsibility.

Did I answer why I chose the current design, how it can be improved from my point of view and why it is not a good idea to move all handle methods into one listener?